core/kernel/tee_ta_manager.c

```diff
@@ -128,10 +129,9 @@ static bool tee_ta_try_set_busy(struct tee_ta_ctx *ctx)
 	return rc;
 }
 
-static void tee_ta_set_busy(struct tee_ta_ctx *ctx)
+static void tee_ta_set_busy(struct tee_ta_ctx *ctx __maybe_unused)
 {
-	if (!tee_ta_try_set_busy(ctx))
-		panic();
+	assert(tee_ta_try_set_busy(ctx));
```

Comment:
If `NDEBUG` is defined `tee_ta_try_set_busy(ctx)` would never be called.
If this is ever triggered it's due to some complex interaction between threads that could be hard to reproduce, better always panic() to be safe.